search-provider/eks-discovery-feed-provider.c

```diff
@@ -220,47 +244,79 @@ typedef enum {
   DISCOVERY_FEED_SET_CUSTOM_TITLE = 1 << 0
 } DiscoveryFeedCustomProps;
 
-static void
-article_card_descriptions_cb (GObject *source,
+static gboolean
+models_and_shards_for_result (EkncEngine   *engine,
+                              const gchar  *application_id,
                               GAsyncResult *result,
-                              gpointer user_data)
+                              GSList       **models,
+                              GSList       **shards,
```

Comment:
It's very weird that `models` transfer a ref but not `shards`. Should be both or neither.